src/lib/mina_base/user_command.ml

```diff
@@ -292,3 +292,17 @@ let filter_by_participant (commands : t list) public_key =
           (Fn.compose
              (Signature_lib.Public_key.Compressed.equal public_key)
              Account_id.public_key))
+
+(* A metric on user commands that should correspond roughly to resource costs
+   for validation/application *)
+let weight : Stable.Latest.t -> int = function
+  | Signed_command signed_command ->
+      Signed_command.payload signed_command |> Signed_command_payload.weight
+  | Parties parties ->
+      Parties.weight parties
+
+(* Fee per weight unit *)
+let fee_per_wu (user_command : Stable.Latest.t) : int =
```

Comment:
could you make weight be of type `Currency.Fee.t`? Or `Currency.Fee.SIgned.t` if we expect negative weight. This is the type we use throughout the code base for fees.

Reply:
I chose to use `int` since I'm expressing a ratio of fees to weight units, rather than a fee.
I could create a separate module for weight units and a module for fee per weight unit, if it's preferable to use measure types.

Reply:
Fixed in 9a6e6c1